app.js

```diff
@@ -235,6 +258,22 @@ function updateDevices(hubSlug){
   }
 }
 
+function getCommandsFromControlGroup(controlGroup){
+  deviceCommands = {}
```

Comment:
Not a JavaScript developer by any stretch of the imagination. I'll incorporate this when I get a chance. There's probably a few other places in the code that could use some cleanup.